File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/ev3lejosv1/listOperationsBasic.java
package generated.main;

import de.fhg.iais.roberta.runtime.*;
import de.fhg.iais.roberta.runtime.ev3.*;

import de.fhg.iais.roberta.mode.general.*;
import de.fhg.iais.roberta.mode.action.*;
import de.fhg.iais.roberta.mode.sensor.*;
import de.fhg.iais.roberta.mode.action.ev3.*;
import de.fhg.iais.roberta.mode.sensor.ev3.*;

import de.fhg.iais.roberta.components.*;

import java.util.LinkedHashSet;
import java.util.HashMap;
import java.util.Set;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

import lejos.remote.nxt.NXTConnection;

public class NEPOprog {
    private static Configuration brickConfiguration;

    private Set<UsedSensor> usedSensors = new LinkedHashSet<UsedSensor>();
    private Hal hal = new Hal(brickConfiguration, usedSensors);

    public static void main(String[] args) {
        try {
             brickConfiguration = new EV3Configuration.Builder()
                .setWheelDiameter(5.6)
                .setTrackWidth(18.0)
                .build();

            new NEPOprog().run();
        } catch ( Exception e ) {
            Hal.displayExceptionWaitForKeyPress(e);
        }
    }

    ArrayList<Float> ___initialEmptyNumbers = new ArrayList<>();
    ArrayList<Boolean> ___initialEmptyBoolean = new ArrayList<>();
    ArrayList<String> ___initialEmptyStrings = new ArrayList<>();
    float ___number = 3;
    boolean ___bool = true;
    String ___string = "c";
    float ___item = 0;

    public void run() throws Exception {
        // Basis List Operations START
        if ( ___initialEmptyNumbers.isEmpty() ) {
            ___initialEmptyNumbers = new ArrayList<>(Arrays.asList((float) 1, (float) 2));
            ___item = ___initialEmptyNumbers.size();
            ___item = ___initialEmptyNumbers.indexOf( (float) 1);
            ___item = ___initialEmptyNumbers.indexOf( (float) 5);
            ___initialEmptyNumbers.set(0, (float) 2);;
        }
        if ( ___initialEmptyBoolean.isEmpty() ) {
            ___initialEmptyBoolean = new ArrayList<>(Arrays.<Boolean>asList(true, false));
            ___item = ___initialEmptyBoolean.size();
            ___item = ___initialEmptyBoolean.indexOf(___bool);
            ___item = ___initialEmptyBoolean.indexOf(null);
            ___initialEmptyBoolean.add(true);;
        }
        if ( ___initialEmptyStrings.isEmpty() ) {
            ___initialEmptyStrings = new ArrayList<>(Arrays.<String>asList("a", "b"));
            ___item = ___initialEmptyStrings.size();
            ___item = ___initialEmptyStrings.indexOf("a");
            ___initialEmptyStrings.set((int) ((___initialEmptyStrings.size() - 1) - 2), "c");;
            ___initialEmptyStrings.add((int) ((___initialEmptyStrings.size() - 1) - 1), "d");;
        }
        // Basis List Operations END
    }

}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/microbitv2/listOperationsBasic.py
import microbit
import random
import math

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

timer1 = microbit.running_time()

___initialEmptyNumbers = []
___initialEmptyBoolean = []
___initialEmptyStrings = []
___number = 3
___bool = True
___string = "c"
___item = 0

def run():
    global timer1, ___initialEmptyNumbers, ___initialEmptyBoolean, ___initialEmptyStrings, ___number, ___bool, ___string, ___item
    # Basis List Operations START
    if not ___initialEmptyNumbers:
        ___initialEmptyNumbers = [1, 2]
        ___item = len( ___initialEmptyNumbers)
        ___item = ___initialEmptyNumbers.index(1)
        ___item = ___initialEmptyNumbers.index(5)
        ___initialEmptyNumbers[0] = 2
    if not ___initialEmptyBoolean:
        ___initialEmptyBoolean = [True, False]
        ___item = len( ___initialEmptyBoolean)
        ___item = ___initialEmptyBoolean.index(___bool)
        ___initialEmptyBoolean.insert(-1, True)
    if not ___initialEmptyStrings:
        ___initialEmptyStrings = ["a", "b"]
        ___item = len( ___initialEmptyStrings)
        ___item = ___initialEmptyStrings.index("a")
        ___initialEmptyStrings[-1 -2] = "c"
        ___initialEmptyStrings.insert(-1 -1, "d")
    # Basis List Operations END

def main():
    try:
        run()
    except Exception as e:
        raise

if __name__ == "__main__":
    main()